components/search/HotelResults/HotelSelect.tsx

```diff
@@ -41,97 +41,121 @@ const HotelSelect: React.FC<HotelSelectUICompleteProps> = ({
+  // We don't want to call the API after clicking one checkbox.
+  // So, in the future we could have an "apply new filters button" if filters change.
+  // Also, store filters in searchParams or localStorage...
+  useEffect(() => {
+    if (!isValid) return;
 
     if (process.env.NODE_ENV === "production") {
       handleFindHotels();
     } else {
-      setLoading(true);
-      setTimeout(() => {
-        setHotelsData({
-          priceRange: {
-            maxPrice: 5000,
-            minPrice: 0,
-          },
-          properties: [
-            {
-              region_id: "1",
-              hotel_id: "1",
-              name: "Hotel 1",
-              image: { description: "img", url: "", alt: "alt" },
-              coordinates: { lat: 0, long: 0 },
-              availability: { available: true, minRoomsLeft: 10 },
-              reviews: { score: 4, totalReviews: 100, starRating: 4 },
-              price: { amount: 100, currency: { code: "USD", symbol: "$" } },
-            },
-            {
-              region_id: "2",
-              hotel_id: "2",
-              name: "Hotel 2",
-              image: { description: "img", url: "", alt: "alt" },
-              coordinates: { lat: 0, long: 0 },
-              availability: { available: true, minRoomsLeft: 10 },
-              reviews: { score: 4, totalReviews: 100, starRating: 4 },
-              price: { amount: 100, currency: { code: "USD", symbol: "$" } },
-            },
-            {
-              region_id: "3",
-              hotel_id: "3",
-              name: "Hotel 3",
-              image: { description: "img", url: "", alt: "alt" },
-              coordinates: { lat: 0, long: 0 },
-              availability: { available: true, minRoomsLeft: 10 },
-              reviews: { score: 4, totalReviews: 100, starRating: 4 },
-              price: { amount: 100, currency: { code: "USD", symbol: "$" } },
-            },
-          ],
-          summary: {
-            matchedPropertiesSize: 3,
-          },
-        });
-        setLoading(false);
-      }, 5000);
+      handleFindHotelsCached();
     }
   }, [region]);
 
   return (
     <div className="container mx-auto p-4">
       <h2 className="text-lg font-semibold mb-4">Available Hotels</h2>
 
+      <div className="flex flex-row gap-4">
+        <button
+          onClick={handleFindHotelsCached}
+          className="mt-4 px-4 py-2 bg-green-500 text-white rounded hover:bg-green-600 disabled:bg-red-500 disabled:cursor-not-allowed"
+          disabled={loading || !region}
+        >
+          {loading ? "Loading..." : "Find Hotels (cached)"}
+        </button>
+        <button
+          onClick={handleFindHotels}
+          className="mt-4 px-4 py-2 bg-blue-500 text-white rounded hover:bg-blue-600 disabled:bg-red-500 disabled:cursor-not-allowed"
+          disabled={loading || !region}
+        >
+          {loading ? "Loading..." : "Find Hotels (actual API data)"}
+        </button>
+      </div>
```

Comment:
We removed it on purpose so you will always see hotels, based on your selected location.